Library/ViewModels/SettingsRequestDataCellViewModel.swift

```diff
@@ -39,16 +39,28 @@ public final class SettingsRequestDataCellViewModel: SettingsRequestDataCellView
       self.awakeFromNibProperty.signal
     ).map(first)
 
+    let userEmailEvent = self.configureWithUserProperty.signal.skipNil()
+      .switchMap { _ in
+        AppEnvironment.current.apiService.fetchGraphUserEmailFields(
+          query: NonEmptySet(Query.user(changeEmailQueryFields())))
+        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+        .materialize()
+    }
+
+     let requestDataAlertText = userEmailEvent.values().map {
+      Strings.Youll_receive_an_email_at_email_when_your_download_is_ready(email: $0.me.email)
+    }
+
     let exportEnvelope = initialUser
       .switchMap { _ in
         AppEnvironment.current.apiService.exportDataState()
           .demoteErrors()
     }
 
-    self.showRequestDataPrompt = exportEnvelope
+    self.showRequestDataPrompt = Signal.combineLatest(exportEnvelope, requestDataAlertText)
+      .filter { $0.0.dataUrl == nil || $0.0.state == .expired || $0.0.expiresAt == nil }
```

Comment:
Since you touched this code, I think we can improve the readability of it by extracting the logic to a new function
```
private func canRequestData(_ envelope: ExportDataEnvelope) -> Bool {
  return envelope.dataUrl == nil || envelope.state == .expired || envelope.expiresAt == nil
}
```
and simply filter using `.filter { canRequestData(0.0) }`